torch/lib/libshm/socket.h

```diff
@@ -111,6 +113,12 @@ class ManagerServerSocket: public Socket {
     }
   }
 
+  void remove() {
+    struct stat st;
+    if (stat(socket_path.c_str(), &st) == 0)
+      SYSCHECK_ERR_RETURN_NEG1(unlink(socket_path.c_str()));
+  }
+
```

Comment:
Before this PR, we don't have a correct file created, so the actual file is never unlinked.
By adding this check, we can have a better debugging result if anything invalid is introduced in the future.